pkg/apiserver/certificate/selfsignedcert_provider.go

```diff
@@ -149,6 +149,7 @@ func (p *selfSignedCertProvider) Run(ctx context.Context, workers int) {
 
 	if p.secretInformer != nil {
 		go p.secretInformer.Run(ctx.Done())
+		cache.WaitForCacheSync(ctx.Done(), p.secretInformer.HasSynced)
```

Comment:
this looks more in line with what we do for other controllers, where we wait for a cache sync before starting queue workers